src/lib.rs

```diff
@@ -150,72 +212,52 @@ impl DisplayManager {
             properties: DisplayManagerProperties::new(),
         }
     }
-    /// Watch for monitor changes.
+
     pub async fn watch_changes(
         manager_obj: Arc<Mutex<DisplayManager>>,
         sway_connection: Arc<Mutex<Connection>>,
-    ) -> Result<(), Box<dyn Error>> {
-        let prefix = "card";
-        let get_monitor_state = |path: &PathBuf| {
-            let mut status = String::from("Off");
-            let status_path = path.join("dpms");
-            if status_path.exists() {
-                File::open(status_path)
-                    .unwrap()
-                    .read_to_string(&mut status)
-                    .unwrap();
-            }
-            let enabled_path = path.join("enabled");
-            let mut enabled = String::from("disabled");
-            if enabled_path.exists() {
-                File::open(enabled_path)
-                    .unwrap()
-                    .read_to_string(&mut enabled)
-                    .unwrap();
-            }
-            (status, enabled)
-        };
-        let get_outputs = || -> Vec<_> {
-            let outputs = fs::read_dir("/sys/class/drm/")
-                .unwrap()
-                .map(|r| r.unwrap())
-                .filter(|item| item.file_name().to_str().unwrap().starts_with(prefix))
-                .map(|item| item.path())
-                .map(|path| {
-                    let status = get_monitor_state(&path);
-                    (path, status)
-                })
-                .collect();
-            outputs
-        };
-        info!("Watching Monitor Changes...");
-        let mut outputs;
+    )-> Result<(), Box<dyn Error>> {
+        let mut prev_monitor_set = HashSet::new();
+        let mut prev_logical_monitor_set = HashSet::new();
         loop {
-            let manager_obj_ref = Arc::clone(&manager_obj);
-            outputs = get_outputs();
-            thread::sleep(Duration::from_millis(600));
-            for output in &*outputs {
-                let curr_status = get_monitor_state(&output.0);
-                if curr_status != output.1 {
-                    debug!("Change Detected");
-                    info!("Displays changed...");
-                    info!("Display Status: {:?}", outputs);
-                    if let Err(e) = manager_obj_ref
-                        .lock()
-                        .await
-                        .get_monitor_info(&sway_connection)
-                        .await
-                    {
-                        error!("{e}");
-                    };
-                    Self::emit_monitors_changed().await?;
-                    info!("Emiting MonitorsChanged signal...");
-                    break;
+            thread::sleep(Duration::from_millis(700));
```

Comment:
Do we expect this value may need to be tuned based on some env factor?  Any way to refactor to not require a sleep?

Reply:
I actually tried listening the output changes directly by watching the `/sys/class/drm/card*/status` file for changes using`inotify`. But this file seemed to change automatically every few milliseconds. I also tried to look for any wayland roots (`wlr`) functions that would allow to achieve the same result, but couldn't find any. I had two options at this point: 
1. Poll `/sys/class/drm/card*/status` 
2. Poll sway IPC interface. 
I ended up choosing the second option because it allowed me to monitor for resolution changes as well.

I still think `wlr` might have the option to listen for changes to output settings.

Reply:
I actually tried listening the output changes directly by watching the `/sys/class/drm/card*/status` file for changes using`inotify`. But this file seemed to change automatically every few milliseconds. I also tried to look for any wayland roots (`wlr`) functions that would allow to achieve the same result, but couldn't find any. I had two options at this point: 
1. Poll `/sys/class/drm/card*/status` 
2. Poll sway IPC interface. 
I ended up choosing the second option because it allowed me to monitor for resolution changes as well.

I was also unhappy with the sleep so I decided to set a large poll interval (700ms) and hoped `regolith-displayd` won't hog up resources in the background which seems to be true for the most part. 

I still think `wlr` might have the option to listen for changes to output settings. I'm ready to get rid of the sleep if something like that exists. m Lmk if you find anything. I'll also take a look at other alternatives for this.